array.c

```diff
@@ -317,12 +317,12 @@ rb_ary_ptr_use_end(VALUE ary)
 #endif
 }
 
+static inline void memfill(register VALUE *mem, register long size, register VALUE val);
```

Comment:
just move `mem_clear` under `memfill` function, no need to duplicate the definition